OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -67,23 +77,36 @@ public override Activity Tick(Actor self)
 				landingInitiated = true;
 			}
 
-			if (!soundPlayed && aircraft.Info.LandingSounds.Length > 0 && !self.IsAtGroundLevel())
+			var altitude = self.World.Map.DistanceAboveTerrain(self.CenterPosition);
+			var landAltitude = self.World.Map.DistanceAboveTerrain(target.CenterPosition + offset) + aircraft.Info.LandAltitude;
```

Comment:
> Carryall dances around instead of landing to drop the cargo (maybe related to my landing position question above?)

Rebasing this on top of #16417 and replacing `aircraft.Info.LandAltitude` with `aircraft.LandAltitude` in the above line fixes this issue for me. (introducing `aircraft.LandAltitude` is also what fixed the issue of loaded carryalls landing below the terrain.)